### DISCUSSION TITLE
useArrayField returns never[] for fields on optional array fields

### DISCUSSION THREAD

**Discussion by kevineaton603** [react-hook-form/react-hook-form#5517] [Q&A] [answered]
**Describe the bug**
If form includes optional array field, `useFieldArray` will return with type `never[]` for `fields`

**To Reproduce**

1. Go to https://codesandbox.io/s/exciting-leavitt-tg416?file=/src/App.tsx
2. Hover over `fields` in `App.tsx`

**Codesandbox link (Required)**
https://codesandbox.io/s/exciting-leavitt-tg416?file=/src/App.tsx

**Expected behavior**
Expected `fields` to be return as type `FieldArrayWithId<FormValues, "questions", "id">[]`

**Screenshots**
![Screen Shot 2021-06-07 at 10 45 51 AM](https://user-images.githubusercontent.com/25751382/121037727-aea7bd00-c77d-11eb-93cd-edff42c30a41.png)

**Additional context**
Seems to effect the other `useFieldArray` methods as well such as `append`. (Inlcuded in Sandbox)

**Comment by JamesWong1999** [comment, marked as answer]
```diff
type FormValues = {
-  questions?: {
+  questions: {
    question: string;
  }[];
};
```

should resolve the problem.
